wordpress-mobile/WordPress-iOS
# Update Global Styles endpoint to support non-experimental paths

This updates the Global styles endpoint to use the non-experimental paths for fetching the block editor settings. 

**Related PR** https://github.com/wordpress-mobile/WordPressKit-iOS/pull/416

## To test:
For each of these cases, it'll be enough to open the editor and validate that the correct color options are available. It would also be useful to test using a network monitoring tool to make sure the correct paths are called. This will help ensure you aren't seeing cached data if you're reusing the same site for testing.

- [ ] Self-Hosted Site with WP 5.8 and Gutenberg 11.1
    - Expect the request to be made with `/wp-block-editor/v1`
- [ ] Self-Hosted Site with Gutenberg version <= 11.0
    - Expect the request to be made with `/wp-block-editor/v1` which will fail, so the fallback of `wp/v2/themes` should be used
- [ ] WPCom or Jetpack connected site with WP 5.8 and Gutenberg 11.0
    - Expect the request to be made with `/wp-block-editor/v1`
    - This will need to be tested with D63704 until that's deployed
- [ ] WPCom or Jetpack connected site with WP 5.8 and Gutenberg version >= 11.1
    - Expect the request to be made with `/wp-block-editor/v1`=
    - This will need to be tested with D63704 until that's deployed

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Update unit tests to validate each flow

## PR submission checklist:
